tcpmaxconn/tcpmaxconn.go

```diff
@@ -4,20 +4,19 @@ import (
 	"bufio"
 	"fmt"
 	"net"
-	"os"
 	"runtime"
 	"sync"
 	"strconv"
 	"time"
 	"flag"
 )
 
-func connection_handler(id int, host string, port int, wg *sync.WaitGroup) {
+func connection_handler(id int, host string, port int, wg *sync.WaitGroup) error{
 	fmt.Println("\t runner "+strconv.Itoa(id)+" is initiating a connection")
 	conn, err := net.Dial("tcp", host+":"+strconv.Itoa(port))
 	if err != nil {
 		fmt.Println(err)
-		os.Exit(1)
+		return err
```

Comment:
its nicer, but... how are we capturing this error in the "parent" goroutine? A destination not reachable means as many retries as number of connections against the destination, or does the program stop when we raise an error than is not captured?

Reply:
I assume that the parent goroutine will see the error, so we should handle it properly (not done now).
We could retry it (introducing some max retries) or we could skip it and leave the max connections without these failed ones, what do you think? (I would go for the retry option)

Reply:
Error management should be an item in the backlog. And it should depend on how we are running the tool (in the mode of "open and keep X connections open", retrying makes sense. In the mode of "try to open up to X connections" it may not have sense)

Reply:
https://github.com/dachad/check-max-tcp-connections/issues/7